avahi-common/defs.h

```diff
@@ -335,7 +340,15 @@ enum {
     AVAHI_DNS_TYPE_MX = 0x0F,
     AVAHI_DNS_TYPE_TXT = 0x10,
     AVAHI_DNS_TYPE_AAAA = 0x1C,
-    AVAHI_DNS_TYPE_SRV = 0x21
+    AVAHI_DNS_TYPE_SRV = 0x21,
+    AVAHI_DNS_TYPE_NAPTR = 0x23,
+    AVAHI_DNS_TYPE_CERT = 0x25,
+    AVAHI_DNS_TYPE_DNAME = 0x27,
+    AVAHI_DNS_TYPE_SSHFP = 0x2c,
+    AVAHI_DNS_TYPE_TLSA = 0x34,
+    AVAHI_DNS_TYPE_OPEPGPKEY = 0x3d,
+    AVAHI_DNS_TYPE_SVCB = 0x40,
+    AVAHI_DNS_TYPE_HTTPS = 0x41,
```

Comment:
> Expand supported qtypes with some new types

> Apple devices have been seen to use HTTPS RR queries even on mDNS. Define that record type to ensure it would be printed nicely in logs.

As far as I can see HTTPS RRs aren't parsed anywhere so they are still dumped according to rfc3597. I'm not sure RRs that aren't fully supported should be included in this list. (At this point only fully supported RRs are listed here).

It could be I'm missing something though. Could you expand on what this PR is trying to improve?

Reply:
I disagree. These changes are enough to print these types without cryptic TYPE65 for known record types. Of course we want more complex decoding for all those types. Especially SVCB and HTTPS have great potential to be very common, so we will need them fixed later.

But that would require non-trivial code. These changes should make a bit better and very safe, without bringing any regressions. But yes, at least mentioning those types do not have avahi_record_to_string implementation in a comment would be nice.

This is how it is decoded in unknown records format:
```
$ echo "IN HTTPS 0 example.org " | named-rrchecker -u 
CLASS1	TYPE65	\# 15 0000076578616D706C65036F726700
```

This is how it should be formatted after my change
```
IN	HTTPS	\# 15 0000076578616D706C65036F726700
```

While this is suboptimal and not fully decoded, it is still much better with at least q-types converted.

Reply:
I don't feel strongly about this but to me it just looks inconsistent to add unsupported types to the list of supported types.

I'm not sure how helpful it is either. For example NSEC RRs are used much more often in the context of mDNS and I haven't seen any feature requests asking to prettify them a bit. I saw bug reports like https://github.com/avahi/avahi/issues/379 though where it's necessary to actually parse NSEC RRs to reflect them properly.

Reply:
Having taken a look at f6712902a92eb82b6c8d7e7fd0980a20a716fa0d I don't think this PR should be merged at all because the idea behind that commit was to format unknown types according to RFC3597 where
> an unknown RR type is
   represented by the word "TYPE" immediately followed by the decimal RR
   type number, with no intervening whitespace

, move all the supported types to `avahi-common/defs.h` and keep stuff like `AVAHI_DNS_TYPE_TKEY` separately.  It's a NACK from me.

Reply:
Oh, in that case we want to include NSEC to printed types as well. Of course reacting on its presence would be more difficult and not handled now.

Reply:
That is not list of supported types, but list of known types. That is somehow important difference.

Reply:
> That is not list of supported types, but list of known types

In the context of avahi it's a list of supported types avahi knows how to parse and display properly. `AVAHI_DNS_TYPE_OPT` isn't in `avahi-common/defs.h` because avahi can't parse it. It is needed to do some things so it's known and kept in `avahi-common/rr.h`.

Other than that
> IN	HTTPS	\# 15 0000076578616D706C65036F726700

violates the RFC.